src/GitObjectDb/Models/ObjectRepositoryContainer{TRepository}.cs

```diff
@@ -23,36 +24,44 @@ namespace GitObjectDb.Models
     public class ObjectRepositoryContainer<TRepository> : ObjectRepositoryContainer, IObjectRepositoryContainer<TRepository>
         where TRepository : class, IObjectRepository
     {
+        private readonly IObjectRepositoryLoader _repositoryLoader;
         private readonly ComputeTreeChangesFactory _computeTreeChangesFactory;
         private readonly ObjectRepositoryMergeFactory _objectRepositoryMergeFactory;
         private readonly ObjectRepositoryRebaseFactory _objectRepositoryRebaseFactory;
-        private readonly IObjectRepositoryLoader _repositoryLoader;
         private readonly IRepositoryProvider _repositoryProvider;
         private readonly GitHooks _hooks;
+        private readonly ILogger<ObjectRepositoryContainer> _logger;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="ObjectRepositoryContainer{TRepository}"/> class.
         /// </summary>
-        /// <param name="serviceProvider">The service provider.</param>
         /// <param name="path">The path.</param>
-        public ObjectRepositoryContainer(IServiceProvider serviceProvider, string path)
+        /// <param name="repositoryLoader">The repository loader.</param>
+        /// <param name="computeTreeChangesFactory">The <see cref="IComputeTreeChanges"/> factory.</param>
+        /// <param name="objectRepositoryMergeFactory">The <see cref="IObjectRepositoryMerge"/> factory.</param>
+        /// <param name="objectRepositoryRebaseFactory">The <see cref="IObjectRepositoryRebase"/> factory.</param>
+        /// <param name="repositoryProvider">The repository provider.</param>
+        /// <param name="hooks">The hooks.</param>
+        /// <param name="logger">The logger.</param>
+        public ObjectRepositoryContainer(string path,
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Constructor has 8 parameters, which is greater than the 7 authorized. ([csharpsquid:S107](https://sonarcloud.io/organizations/frblondin-github/rules#rule_key=csharpsquid%3AS107))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=GitObjectDb&pullRequest=33&issues=AWe4KJcBD1JUB5M3-M2g&open=AWe4KJcBD1JUB5M3-M2g)

[ISSUE_KEY]: AWe4KJcBD1JUB5M3-M2g